app/controllers/courses/github_repos_controller.rb

```diff
@@ -0,0 +1,37 @@
+# This controller is nested in the routing hierarchy as a child controller of 
+# courses_controller
+# see this tutorial for details on how we do this https://gist.github.com/jhjguxin/3074080
+require 'Octokit_Wrapper'
+
+module Courses
+  class GithubReposController < ApplicationController
+    layout 'courses'
+    before_action :load_parent
+    before_action :set_github_repo, only: [:show]
+
+    load_and_authorize_resource :course
+    load_and_authorize_resource :github_repo, through: :course
+
+    def index
+      @github_repos = @parent.github_repos.all
+    end
+
+    def show
+    end
+
+    private
+      # Use callbacks to share common setup or constraints between actions.
+      def set_github_repo
+        @github_repo = GithubRepo.find(params[:id])
+      end
+
+      def machine_user
```

Comment:
This isn't necessary, `ApplicationController`, the parent class of all the controllers in the app exposes a `github_machine_user` which you can use everywhere

Reply:
Also, is it being used in this controller for anything?